src/arr/compiler/pyret.arr

```diff
@@ -145,31 +153,38 @@ fun main(args):
               display-progress: display-progress
             })
           failures = filter(CS.is-err, result.loadables)
-          when is-link(failures):
+          if is-link(failures):
             for each(f from failures) block:
               for lists.each(e from f.errors) block:
                 print-error(tostring(e))
                 print-error("\n")
               end
-              raise("There were compilation errors")
+              _ = print("There were compilation errors\n")
+              failure-code
             end
+          else:
+            success-code
           end
         else if r.has-key("run"):
-          CLI.run(r.get-value("run"), CS.default-compile-options.{
+          _ = CLI.run(r.get-value("run"), CS.default-compile-options.{
               standalone-file: standalone-file,
               compile-module: true,
               display-progress: display-progress,
               check-all: check-all
             })
+          success-code
         else:
-          print(C.usage-info(options).join-str("\n"))
-          raise("Unknown command line options")
+          _ = print(C.usage-info(options).join-str("\n"))
+          _ = print("Unknown command line options\n")
+          failure-code
```

Comment:
In these cases, I wonder if we should just keep the "raise" around, and have the toplevel's behavior for uncaught raises be to set the exit code and show the exception.  In particular, just printing here gives _less_ information, because the `raise` would collect and print out a stack trace to aid debugging.